lib/test.js

```diff
@@ -88,6 +89,20 @@ function Test (options) {
   Readable.apply(this, options)
 }
 
+Test.prototype.beforeEach = function (fn) {
+  this.on('beforechild', fn)
+}
+
+Test.prototype.afterEach = function (fn) {
+  this.on('afterchild', fn)
+}
+
+Test.prototype.afterAll = function (fn) {
+  this.on('end', fn)
+}
+
+Test.prototype.tearDown = Test.prototype.afterAll
```

Comment:
Could tearDown instead accept an optional `done` method?

* if `done` is provided, test doesn't end and wait for for it to be executed.
* if no `done` parameter is expected in the closure provided, fallback to the `autoend` logic? -- would this potentially make it easier to implement mocha-style hooks?